commons/src/main/java/com/powsybl/commons/json/JsonUtil.java

```diff
@@ -217,6 +217,52 @@ public static void writeOptionalIntegerField(JsonGenerator jsonGenerator, String
         return notFound;
     }
 
+    public static <T extends Extendable> List<Extension<T>> updateExtensions(JsonParser parser, DeserializationContext context, T toUpdatedParameters) throws IOException {
+        return updateExtensions(parser, context, SUPPLIER.get(), null, toUpdatedParameters);
+    }
+
+    public static <T extends Extendable> List<Extension<T>> updateExtensions(JsonParser parser, DeserializationContext context,
+                                                                             ExtensionProviders<? extends ExtensionJsonSerializer> supplier, T toUpdatedParameters) throws IOException {
+        return updateExtensions(parser, context, supplier, null, toUpdatedParameters);
+    }
+
+    public static <T extends Extendable> List<Extension<T>> updateExtensions(JsonParser parser, DeserializationContext context,
+                                                                             ExtensionProviders<? extends ExtensionJsonSerializer> supplier, Set<String> extensionsNotFound, T toUpdatedParameters) throws IOException {
+        Objects.requireNonNull(parser);
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(supplier);
+        Objects.requireNonNull(toUpdatedParameters);
+
+        List<Extension<T>> extensions = new ArrayList<>();
+        while (parser.nextToken() != JsonToken.END_OBJECT) {
+            Extension<T> extension = updateExtension(parser, context, supplier, extensionsNotFound, toUpdatedParameters);
+            if (extension != null) {
+                extensions.add(extension);
+            }
+        }
+        return extensions;
+    }
+
+    public static <T extends Extendable, E extends Extension<T>> E updateExtension(JsonParser parser, DeserializationContext context,
+                                                                                   ExtensionProviders<? extends ExtensionJsonSerializer> supplier, Set<String> extensionsNotFound, T toUpdateExtensions) throws IOException {
+        String extensionName = parser.getCurrentName();
+        ExtensionJsonSerializer<T, E> extensionJsonSerializer = supplier.findProvider(extensionName);
+        if (extensionJsonSerializer != null) {
+            parser.nextToken();
+            if (toUpdateExtensions.getExtensionByName(extensionName) != null) {
+                return extensionJsonSerializer.deserializeAndUpdate(parser, context, (E) toUpdateExtensions.getExtensionByName(extensionName));
+            } else {
+                return extensionJsonSerializer.deserialize(parser, context);
+            }
+        } else {
+            if (extensionsNotFound != null) {
+                extensionsNotFound.add(extensionName);
+            }
+            skip(parser);
+            return null;
+        }
+    }
+
```

Comment:
I think it can be refactored with `readExtensions` and `readExtension`, it looks similar.